### ISSUE TITLE
Mat/Pat - Send email "Reminder to return to work

### ISSUE THREAD

**Issue by bhumin-fw** [bcgov/digital-journeys#1218]
As a user I want to setup workflow that send and email to applicant about leave end. This content was copied from ticket [#874](https://app.zenhub.com/workspaces/digital-journeys-61c10fe84c8bcb001491c5f5/issues/gh/bcgov/digital-journeys/874).

We have leave start date logic in place. Now need to add leave end date logic (last date of maternity and/or parental leave). Based on end date needs to calculate date and time to send this email.

Please find email detail as per below.

> **4. Reminder to return to work.**
> 
> Sent to applicant's alternate email. To be sent 6 weeks before end of leave (last date of maternity and/or parental leave).
> 
> **Sub: Reminder to notify return to work 4 weeks before Maternity/Parental Leave **
> This is to inform you that you must inform your supervisor about your intent to return to work at least 4 weeks before your intended date of return. Please include the following information i your communication to your supervisor
> - Type of leave you took
> - First day of work
> - Phone number for supervisor to connect with you if needed
> You must ask your supervisor to submit an AskMyHR service request for "My Team or Organization" under the following category "Leave & Time Off > Maternity, Parental, Adoption" with the letter attached.
> 
> You can also inform your supervisor about your intent to extend your leave if you do not intend to return to work on your originally chosen return date. To extend your leave, your should ask your supervisor to update your Maternity and Parental leave form with the new length of maternity or parental leave and resubmit the form.
> 
> [only if deferred allowance] (Copy To be confirmed)
> To claim your allowance after you return to work, you should have applied for Employment Insurance after you started your leave. You will need a copy of your proof of approval. To get your proof of approval -
> 
> Log into the Service Canada website. Click on “My Latest Claim".
> Take a picture or screenshot of the contents on "My Latest Claim" page. This information confirms that the you have been approved by Service Canada to receive the Maternity/Parental benefits.
> [only if taking extended parental leave allowance] Since you opted for extended parental leave allowance, you must provide another proof of approval for your parental leave allowance. You should complete the above steps again after you start your parental leave to claim parental leave allowance after you return.

Task:
- [ ] Setup logic on form and workflow to determine time to send email(six week before enddate)
- [ ] Setup send mail task in workflow

**Comment by MeghanStothers** [comment]
Bhumin to check with Iman and implement a solution.

**Comment by MeghanStothers** [comment]
Bhumin to add testing details once Iman posts changes

**Comment by bhumin-fw** [comment]
To test out this ticket,
we need to set date of tomorrow or near future for "reminderEmailDate".
That can be done in 2 ways.
1. Select enddate 4 week after tomorrow. To do that please select appropriate start date.
verify it in console. Open console by right click on form page - > select "inspect" -> select "network" tab
Select latest draft request and go to payload tab, in this click on "data" and expand it. checkout below image for more details.
make sure "reminderEmailDate" set to tomorrow ("YYYY-MM-DD" format)
![Screenshot 2023-08-09 at 10.51.10 AM copy.png](https://images.zenhubusercontent.com/62e98579ff9c075848ccdb67/ff976ba5-33a8-437b-9372-67048202dd65)


2. In this approach we are setting "reminderEmailDate" manually in form.
go to form -> edit -> find "reminderEmailDate" hidden field -> select "data" tab -> go to "calculated value" section.
here please comment out all code. and write line
`value = "2023-08-10";`
here date is in "YYYY-MM-DD" format. please add any date in future.
Note: Make sure you enable comment code again and delete additional line once you submit form.

**Comment by MeghanStothers** [comment]
@Bhumin - thanks for the details. Was your test successful? For next steps, should the PSA dev team test this one?

**Comment by bhumin-fw** [comment]
Yes Meghan. It is available for PSA team to test it on DEV and TEST environment.

**Comment by chriscaldwell-psa** [comment]
In the Dev env, I sent 3 Mat/pat forms for Bhumin to review. I will check with him on Monday if the testing was OK.

**Comment by bhumin-fw** [comment]
@chriscaldwell-psa I have fix issue in TEST environment. Now you should receive email for the same.
Let me know if you still haven't received any email. We can connect tomorrow after standup for 5 minute and submit new form. I'll review the same in system and confirm and we will for email.

**Comment by ayushdamani** [comment]
Submitted a form on Friday but did not receive a reminder email. Bhumin is looking into this.

**Comment by bhumin-fw** [comment]
@ayushdamani 
It is ready to test on DEV and TEST. Please submit new form and do let me know (in which env you have submitted form), so I can update date in workflow and you will receive an email by tomorrow.

**Comment by ayushdamani** [comment]
Received the email - a reminder to return to work ✅

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBdmM3IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--d802530f17e73911b2b6dc2572e4945945b80667/image.png)
